validator/src/domain/event.rs

```diff
@@ -0,0 +1,35 @@
+use std::collections::HashMap;
+
+use serde::{Deserialize, Serialize};
+
+use domain::{AdUnit, BigNum};
+
+#[serde(tag = "type", rename_all = "SCREAMING_SNAKE_CASE")]
+#[derive(Serialize, Deserialize)]
+pub enum Event {
+    Impression {
+        publisher: String,
+    },
+    #[serde(rename_all = "camelCase")]
+    UpdateImpressionPrice {
+        publisher: String,
+        price: u64,
+        ad_unit: AdUnit,
+    },
+    ImpressionWithCommission {
+        earner: Vec<Earner>,
+    },
+    ImpressionPricePerCase {},
+    Pay {
```

Comment:
It appears that Pay & PauseChannel & Close?!? also require adUnit in the JS impl?

Reply:
that's wrong, please PR to the JS impl or discuss with Sam

Reply:
@samparsky which types should have the `AdUnit`?